bench/src/main.rs

```diff
@@ -163,36 +47,40 @@ async fn bench(
     rpc_client: Arc<RpcClient>,
     tx_count: usize,
     funded_payer: Keypair,
-    seed: u64,
+    seed: Option<u64>,
     block_hash: Arc<RwLock<Hash>>,
     current_slot: Arc<AtomicU64>,
     tx_metric_sx: UnboundedSender<TxMetricData>,
     log_txs: bool,
-    transaction_size: TransactionSize,
+    tx_size: TxSize,
 ) -> Metric {
     let map_of_txs: Arc<DashMap<Signature, TxSendData>> = Arc::new(DashMap::new());
     // transaction sender task
     let api_caller_result = {
         let map_of_txs = map_of_txs.clone();
         let rpc_client = rpc_client.clone();
         let current_slot = current_slot.clone();
+
         tokio::spawn(async move {
             let map_of_txs = map_of_txs.clone();
-            let n_chars = match transaction_size {
-                TransactionSize::Small => 10,
-                TransactionSize::Large => 240, // 565 is max but we need to lower that to not burn the CUs
+
+            let n_chars = match tx_size {
+                TxSize::Small => 10,
+                TxSize::Large => 240, // 565 is max but we need to lower that to not burn the CUs
             };
-            let rand_strings = BenchHelper::generate_random_strings(tx_count, Some(seed), n_chars);
+
+            let mut rng = BenchHelper::create_rng(seed);
+            let rand_strings = BenchHelper::generate_random_strings(&mut rng, tx_count, n_chars);
 
             let bench_start_time = Instant::now();
 
             for rand_string in &rand_strings {
                 let blockhash = { *block_hash.read().await };
-                let tx = match transaction_size {
-                    TransactionSize::Small => {
+                let tx = match tx_size {
+                    TxSize::Small => {
                         BenchHelper::create_memo_tx_small(rand_string, &funded_payer, blockhash)
```

Comment:
Can have a single instruction create_memo_tx with number of accounts as argument

Reply:
Fixed in a new function